src/Text/Gigaparsec/Char.hs

```diff
@@ -90,15 +91,22 @@ char c = satisfy (== c)
 
 @since 0.1.0.0
 -}
-satisfy :: (Char -> Bool) -- ^ the predicate, @test@, to test the next character against, should one
+satisfy :: (Char -> Bool) -- ^ the predicate, @pred@, to test the next character against, should one
                           -- exist.
-        -> Parsec Char    -- ^ a parser that tries to read a single character @c@, such that @test c@
+        -> Parsec Char    -- ^ a parser that tries to read a single character @c@, such that @pred c@
                           -- is true, or fails.
 satisfy test = Internal.Parsec $ \st ok err ->
   case Internal.input st of
     (x: xs) | test x  ->
-      ok x (st { Internal.input = xs, Internal.consumed = True })
+      ok x ((updatePos st x) { Internal.input = xs, Internal.consumed = True })
     _                 -> err st
+  where
+    updatePos st x =
+      let (l, c) = updatePos' (Internal.line st, Internal.col st) x
+      in  st { Internal.line = l, Internal.col = c }
+    updatePos' (l, _) '\n' = (l + 1, 1)
+    updatePos' (l, c) '\t' = (l, ((c + 3) .&. (-4)) .|. 1)
+    updatePos' (l, c) _    = (l, c + 1)
```

Comment:
`updatePos` in `satsify` unpacks the line even if it won't update it. Is this inefficient?